modules/nlu/src/backend/engine.ts

```diff
@@ -125,28 +126,23 @@ export default class Engine implements NLUEngine {
     let model: Partial<UntrainedModel> = {
       startedAt: new Date(),
       languageCode: input.languageCode,
+      success: false,
       data: {
         input
       }
     }
 
     try {
       const output = await Trainer(input, Engine.tools)
-      if (!output) {
-        model.success = false
-      } else {
-        _.merge(model, { success: true, data: { output } })
-      }
+      model.data.output = output
+      model.success = true
     } catch (err) {
-      model.success = false
-      this.logger.error(`Could not finish training NLU model : ${err}`)
+      this.logger.attachError(err).error(`Could not finish training NLU model : ${err}`)
     } finally {
       model.finishedAt = new Date()
-      model = model as Model
-    }
-
-    if (!trainAllCtx) {
-      model = this._mergeModels(previousModel, model as Model)
+      if (!trainAllCtx) {
+        model = this._mergeModels(previousModel, model as Model)
```

Comment:
I wanted you to change this function (`_mergeModels`) to mergeOutput

Reply:
Done in next PR